src/frame/mod.rs

```diff
@@ -12,7 +12,80 @@ use std::{borrow::Cow, error, fmt};
 use crate::bytes::Bytes;
 
 /// A Modbus function code is represented by an unsigned 8 bit integer.
-pub type FunctionCode = u8;
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FunctionCode {
+    ReadCoils,                  //=> 0x01,
+    ReadDiscreteInputs,         // => 0x02,
+    WriteSingleCoil,            //(_, _) => 0x05,
+    WriteMultipleCoils,         //(_, _) => 0x0F,
+    ReadInputRegisters,         //(_) => 0x04,
+    ReadHoldingRegisters,       //(_) => 0x03,
+    WriteSingleRegister,        //(_, _) => 0x06,
+    WriteMultipleRegisters,     //(_, _) => 0x10,
+    MaskWriteRegister,          //(_, _, _) => 0x16,
+    ReadWriteMultipleRegisters, //(_) => 0x17,
+    Custom(u8),                 // => code,
+}
+
+impl From<FunctionCode> for u8 {
```

Comment:
The actual conversions should be implemented by a `const fn` method.

Reply:
done